gcloud-java-dns/src/main/java/com/google/cloud/dns/testing/LocalDnsHelper.java

```diff
@@ -273,13 +281,18 @@ private String toJson(String message) throws IOException {
   private class RequestHandler implements HttpHandler {
 
     private Response pickHandler(HttpExchange exchange, CallRegex regex) {
-      URI relative = BASE_CONTEXT.relativize(exchange.getRequestURI());
+      URI relative = null;
+      try {
+        relative = BASE_CONTEXT.relativize(new URI(exchange.getRequestURI().getRawPath()));
```

Comment:
I assume that you are talking about the Batch request itself (POST /batch) and not the individual batch calls that this emulator is triggering. Right? If so, why not to check for if `getPath()` for `new URI(exchange.getRequestURI().getRawPath()))` (which you can save in a local variable) is equals to "/batch" and in that case assign relative to it without applying `relativize` on it?